pkg/kubelet/kubelet_pods.go

```diff
@@ -1768,6 +1764,36 @@ func (kl *Kubelet) determinePodResizeStatus(pod *v1.Pod, podStatus *v1.PodStatus
 	return podResizeStatus
 }
 
+func (kl *Kubelet) hasPodInitialized(pod *v1.Pod, status *v1.PodStatus) bool {
+	if len(pod.Spec.InitContainers) == 0 {
+		return true
+	}
+
+	if status == nil {
+		status = &pod.Status
+	}
+
+	_, readyToStartContainersCondition := podutil.GetPodCondition(status, v1.PodReadyToStartContainers)
```

Comment:
ReadyToStartContainers is still in beta so you may need to check for feature gate here.

If feature gate is not present, this will be undefined.

Reply:
Please add some unit tests covering if PodReadyToStart gate was turned off.

Reply:
added unit tests for this function.